test/Microsoft.AspNet.Mvc.TagHelpers.Test/LinkTagHelperTest.cs

```diff
@@ -298,12 +287,11 @@ public static TheoryData RunsWhenRequiredAttributesArePresent_Data
             helper.Process(context, output);
 
             // Assert
-            Assert.Null(output.TagName);
-            Assert.NotNull(output.Content);
-            Assert.True(output.IsContentModified);
+            Assert.True(output.IsContentModified || !string.IsNullOrEmpty(output.PostElement.GetContent()));
         }
 
         [Fact]
+        // Does not work as expected - https://github.com/aspnet/Mvc/issues/2639
```

Comment:
Remove this comment since the test executes successfully. We know fixing #2639 will involve changing tests. But may be worth changing the test name to something like `PreservesOrderOfNonHrefAttributes`.